integration/update_compat_test.go

```diff
@@ -338,93 +420,154 @@ func testUpdateCurrentDateCompat(t *testing.T, testCases map[string]updateCurren
 			t.Parallel()
 
 			// Use per-test setup because updates modify data set.
-			ctx, targetCollections, compatCollections := setup.SetupCompat(t)
+			s := setup.SetupCompatWithOpts(t, &setup.SetupCompatOpts{
+				Providers: shareddata.AllProviders(),
+			})
 
-			update := tc.update
-			require.NotNil(t, update, "`update` must be set")
+			nonEmptyResults := testUpdateCurrentDateCollections(t, s, updateCurrentDateCollectionParams{
+				paths:  tc.paths,
+				update: tc.update,
+				filter: tc.filter,
+			})
 
-			paths := tc.paths
+			assert.True(t, nonEmptyResults, "expected non-empty results (some documents should be modified)")
+		})
+	}
+}
 
-			var nonEmptyResults bool
-			for i := range targetCollections {
-				targetCollection := targetCollections[i]
-				compatCollection := compatCollections[i]
-				t.Run(targetCollection.Name(), func(t *testing.T) {
-					t.Helper()
+// testUpdateCurrentDateCollections applies update operation to
+// the collections and returns true if any collection was modified.
+// It checks current date in compat and target are within acceptable difference.
+func testUpdateCurrentDateCollections(t *testing.T, s *setup.SetupCompatResult, p updateCurrentDateCollectionParams) bool {
```

Comment:
`testUpdateCurrentDateCollections` captures common part of `testUpdateCurrentDateCompat` and `testUpdateCurrentDateCompatUnchanged`